### DISCUSSION TITLE
[V2] Svelte router issues

### DISCUSSION THREAD

**Discussion by Gaz492** [wailsapp/wails#839] [Q&A]
#####################################################
     **If you have a technical issue, please do not open a bug this way!** 
                      Please use the `wails issue` command!
          If you do not do this then the issue may be closed automatically.                      

 NOTE: If your bug is related to Windows, make sure you read 
 the [Windows Developer Guide](https://wails.app/guides/windows/)
#####################################################

**Description**
When using svelte-navigator to add routes/different pages the built app will try and load a file on the route instead of using the route.
E.g I have a route named about which is accessed by clicking on a button that links to /about, instead of loading the route it tries to load a file at /about

**To Reproduce**
Steps to reproduce the behaviour:
1. Create an app using the svelte template
2. Install and setup svelte-navigator routes
3. Build app and run it
4. Click on a router link

**Expected behaviour**
Should load the router link and not try and load a file

**Screenshots**

**System Details**
Please provide your platform, GO version and variables, etc
go version go1.17.1 windows/amd64
Wails v2 beta 3

**Additional context**
While playing around with the alpha version of v2 I did not run into this issue

- [x] This issue is for Windows and I have read the [Windows Developer Guide](https://wails.app/guides/windows/)

**Comment by leaanthony** [comment]
That's sounds like a svelte-router issue: if it issues a navigate request then it's going to try and load a page. Let me know if you have more information.

**Comment by leaanthony** [comment]
Have you tried hash-based routing?

**Reply by Redid** [reply]
Moved from #908 :)

I've tried both [Svelte Hash](https://github.com/mefechoel/svelte-navigator/tree/main/example/custom-hash-history) and Svelte built-in memory history without luck.

Also as mentioned in #908 I've tried different lightweight library called petite-vue and `@click` didn't work directly in Wails as well, which is similar to vue's `ref`, which works properly.

**Reply by leaanthony** [reply]
I'm not really sure how I can help you. Try searching the Microsoft Webview2 feedback repo in GitHub and see why WebView2 might not be executing that as you'd expect.

**Reply by leaanthony** [reply]
Following up...have you tried https://github.com/pynnl/svelte-hash-router? I know hash based routing is supported in web views: https://github.com/wailsapp/wails/issues/917. Please update us with any progress and we'll add to the docs.

**Reply by ianmjones** [reply]
Just to highlight another hash based router option... I've been using https://github.com/ItalyPaleAle/svelte-spa-router on a fairly complex Svelte app with great success and heartily recommend it. I haven't used it with Wails yet, but can't think of a reason for it not to work.

**Reply by leaanthony** [reply]
Webviews generally support hash based routing and we've had success using them for vue, react and angular
